test_responses.py

```diff
@@ -132,14 +132,31 @@ def run():
     [
         ("http://example.com/one", re.compile(r"http://example\.com/one")),
         (re.compile(r"http://example\.com/one"), "http://example.com/one"),
+        (
+            Response(method=responses.GET, url="http://example.com/one"),
+            Response(method=responses.GET, url="http://example.com/two"),
+        ),
     ],
 )
 def test_replace_error(original, replacement):
     @responses.activate
     def run():
-        responses.add(responses.GET, original)
-        with pytest.raises(ValueError):
-            responses.replace(responses.GET, replacement)
+        if isinstance(original, BaseResponse):
+            responses.add(original)
+        else:
+            responses.add(responses.GET, original)
+
+        with pytest.raises(ValueError) as excinfo:
+            if isinstance(replacement, BaseResponse):
+                responses.replace(replacement)
+            else:
```

Comment:
This test is getting complicated. It might be simpler to split out these object based tests into a separate test.

Reply:
I created a separate test function to test replace error on `Response` object.